src/main/java/dan200/computercraft/shared/peripheral/generic/data/ItemData.java

```diff
@@ -116,6 +118,28 @@ private static ITextComponent parseTextComponent( @Nonnull INBT x )
         }
     }
 
+    /**
+     * Retrieve all item groups an item stack pertains to.
+     *
+     * @param stack     Stack to analyse
+     * @return A filled list that contains pairs of item group IDs and their display names.
+     */
+    @Nonnull
+    private static List<Map<String, Object>> getAllItemGroups( @Nonnull ItemStack stack )
+    {
+        ArrayList<Map<String, Object>> groups = new ArrayList<>( 1 );
+
+        for ( ItemGroup g : stack.getItem().getCreativeTabs() )
+        {
+            Map<String, Object> group = new HashMap<>( 2 );
+            group.put( "id", g.getId() );
```

Comment:
It feels like it's probably more useful to access expose the `langId` rather than the numerical ID - AFAIK the latter can change between runs.